From a753ed5b92fee1f6966226950216eeadee65f6bd Mon Sep 17 00:00:00 2001
From: Damiano Galassi <damiog@gmail.com>
Date: Sat, 20 Apr 2024 10:34:09 +0200
Subject: [PATCH] contrib: fix a crash that happens when multiple metadata of
 the same type are appended. This happens when there is both a Dolby Vision
 RPU and HDR10+ payload.

---
 contrib/svt-av1/A02-fix-metadata-size.patch | 26 +++++++++++++++++++++
 1 file changed, 26 insertions(+)
 create mode 100644 contrib/svt-av1/A02-fix-metadata-size.patch

diff --git a/contrib/svt-av1/A02-fix-metadata-size.patch b/contrib/svt-av1/A02-fix-metadata-size.patch
new file mode 100644
index 000000000..91292373b
--- /dev/null
+++ b/contrib/svt-av1/A02-fix-metadata-size.patch
@@ -0,0 +1,26 @@
+From b448e6f43fffcb966296632db32081f3c91e10e3 Mon Sep 17 00:00:00 2001
+From: Damiano Galassi <damiog@gmail.com>
+Date: Sat, 20 Apr 2024 10:28:26 +0200
+Subject: [PATCH] Sum the size of all the metadata if there is more than one of
+ the same type.
+
+---
+ Source/Lib/Encoder/Globals/EbMetadataHandle.c | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/Source/Lib/Encoder/Globals/EbMetadataHandle.c b/Source/Lib/Encoder/Globals/EbMetadataHandle.c
+index d9002ee1..5f29dccd 100644
+--- a/Source/Lib/Encoder/Globals/EbMetadataHandle.c
++++ b/Source/Lib/Encoder/Globals/EbMetadataHandle.c
+@@ -123,7 +123,7 @@ EB_API size_t svt_metadata_size(SvtMetadataArrayT *metadata, const EbAv1Metadata
+         for (size_t i = 0; i < metadata->sz; i++) {
+             SvtMetadataT *current_metadata = metadata->metadata_array[i];
+             if (current_metadata && current_metadata->payload && current_metadata->type == type) {
+-                sz = current_metadata->sz + 1 //obu type
++                sz += current_metadata->sz + 1 //obu type
+                     + 1 //trailing byte
+                     + 1 //header size
+                     + 1; //length field size
+-- 
+2.39.3 (Apple Git-146)
+
-- 
2.34.1

